tutorials/pretrain_tinyllama.md

```diff
@@ -0,0 +1,127 @@
+
+(this table was sourced from the author's [README](https://github.com/jzhang38/TinyLlama/))
+
+
+## Download datasets
+
+You can download the data using git lfs:
+
+```bash
+# Make sure you have git-lfs installed (https://git-lfs.com):
+git lfs install
+```
+
+```bash
+git clone https://huggingface.co/datasets/cerebras/slimpajama-627b data/slimpajama-raw
+git clone https://huggingface.co/datasets/bigcode/starcoderdata data/starcoderdata-raw
+```
+Around 1.2 TB of disk space is required to store both datasets.
+
+## Prepare the datasets for training
+
+In order to start pretraining lit-gpt on it, you need to read, tokenize, and write the data in binary chunks. This will leverage our `lightning.data` optimization pipeline and streaming dataset that comes with Lightning. You will need to have the tokenizer config available:
+
+```bash
+pip install huggingface_hub sentencepiece
+
+python scripts/download.py \
+   --repo_id meta-llama/Llama-2-7b-hf \
+   --access_token your_hf_token
+```
+
+Then, run the preprocessing script for each dataset and split.
+You will require **1.1 TB** of disk space for Starcoder and **2.5** TB of space for the SlimPajama dataset.
+
+**Starcoder:**
+```bash
+python scripts/prepare_starcoder.py \
+  --input_dir data/starcoderdata-raw \
+  --output_dir data/starcoder \
+  --tokenizer_path checkpoints/Llama-2-7b-hf \
+```
+
+**SlimPajama:**
+```bash
+python scripts/prepare_slimpajama.py \
+  --input_dir data/slimpajama-raw/validation \
+  --output_dir data/slimpajama/val \
+  --tokenizer_path checkpoints/Llama-2-7b-hf \
+
+python scripts/prepare_slimpajama.py \
+  --input_dir data/slimpajama-raw/test \
+  --output_dir data/slimpajama/test \
+  --tokenizer_path checkpoints/Llama-2-7b-hf \
+
+python scripts/prepare_slimpajama.py \
+  --input_dir data/slimpajama-raw/train \
+  --output_dir data/slimpajama/train \
+  --tokenizer_path checkpoints/Llama-2-7b-hf \
+```
+
+If you want to run on a small slice of the datasets first, pass the flag `--fast_dev_run=true` to the commands above.
+In the above we are assuming that you will be using the same tokenizer as used in LlaMA/TinyLlama, but any trained [SentencePiece](https://github.com/google/sentencepiece) tokenizer with a 32000 vocabulary size will do here.
+
+
+## Pretraining
+
+Running the pretraining script with its default settings requires at least 8 A100 GPUs.
+
+```bash
+python pretrain/tinyllama.py
+```
+
+The script will save checkpoints periodically to the folder `out/`.
+By default, the `pretrain/tinyllama.py` script will pretrain the Llama 2 7B model with FSDP in
+`bfloat16` mixed precision and gradient accumulation.
+
+You can easily change the size of the model by passing a different string to the model name variable
+
+```python
+model_name = "tiny-llama-1.1b"
+```
+
+at the top of this script.
+
+The currently supported model names are contained in the [config.py](https://github.com/Lightning-AI/lit-gpt/lit_gpt/config.py) file.
```

Comment:
A bit confusing for me.
So this tutorial is for `tinyllama`, but I can use any model that is defined inside config file?

Reply:
I personally tried a couple and works fine. To be honest there is no reason not to. The extras in tinyllama compared to lit-gpt are about the xformer/fusion libraries that speed up things.

Reply:
Right now the main part that's specific to tinyllama is the dataset configuration yes, and the use of lightning.data streaming dataset. I'd like to have a V1 to confirm everything works with these separate scripts and guide. I can adjust the sentence there, it's just mainly copied from the redpajama training guide.